pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -145,6 +145,23 @@ func (p *staticPolicy) validateState(s state.State) error {
 				cID, cset.String(), tmpDefaultCPUset.String())
 		}
 	}
+
+	// 3. It's possible that the set of available CPUs has changed since
+	// the state was written. This can be due to for example
+	// offlining a CPU when kubelet is not running. If this happens,
+	// CPU manager will run into trouble when later it tries to
+	// assign non-existent CPUs to containers. Validate that the
+	// topology that was received during CPU manager startup matches with
+	// the set of CPUs stored in the state.
+	totalKnownCPUs := tmpDefaultCPUset.Clone()
+	for _, cset := range tmpAssignments {
+		totalKnownCPUs = totalKnownCPUs.Union(cset)
+	}
+	if !totalKnownCPUs.Equals(p.topology.CPUDetails.CPUs()) {
+		return fmt.Errorf("current set of available CPUs \"%s\" doesn't match with CPUs in state \"%s\"",
```

Comment:
Could you please add a unit test for each case of this conditional? It should be possible to either extend the existing test for Start or create a new test for this validateState function.

Reply:
I added two test cases now. The test where the amount of CPUs equals in state and topology was already present ("non-corrupted state"), so I just tested the cases where the number of CPUs doesn't match.